### PR TITLE
ES2022 Class fields and static blocks

### PR DESCRIPTION
**Pull request by lahma**
* https://github.com/tc39/proposal-class-fields
* https://github.com/tc39/proposal-private-fields-in-in
* https://github.com/tc39/proposal-class-static-block

3915 new test cases passing.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,13 +97,13 @@ The entire execution engine was rebuild with performance in mind, in many cases
 
 #### ECMAScript 2022
 
-- ❌ Class Fields
+- ✔ Class Fields
```
Comment (sebastienros) [inline README.md:100]: I jumped straight to this line

File: Jint/Jint.csproj [hunk 2]
```diff
@@ -12,9 +12,13 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Esprima" Version="3.0.0-rc-01" />
+    <PackageReference Include="Esprima" Version="3.0.0-rc-02" />
     <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.1.1" PrivateAssets="all" />
     <PackageReference Include="Nullable" Version="1.3.1" PrivateAssets="all" />
+    <PackageReference Include="PolySharp" Version="1.13.1">
```
Comment (sebastienros) [inline Jint/Jint.csproj:18]: 👍

Comment (lahma) [reply Jint/Jint.csproj:18]: This is also an interesting option as it adds extension method too to types: https://github.com/SimonCropp/Polyfill

**Comment by lahma** [comment]
@sebastienros can you update API key for Jint too? https://github.com/sebastienros/jint/actions/runs/4926629275/jobs/8802463465

**Comment by lahma** [comment]
@Jither objects got some very secret and private fields. Not sure if you want to access them debugging-wise in the future.

**Comment by Jither** [comment]
> @Jither objects got some very secret and private fields. Not sure if you want to access them debugging-wise in the future.

Will see when I have time 😊 Not sure what e.g. the V8 debugger does for them (are class fields even implemented in browsers yet?) - will probably want to keep it similar to that, still.

**Comment by lahma** [comment]
> > @Jither objects got some very secret and private fields. Not sure if you want to access them debugging-wise in the future.
> 
> Will see when I have time 😊 Not sure what e.g. the V8 debugger does for them (are class fields even implemented in browsers yet?) - will probably want to keep it similar to that, still.

This is from MS Edge:

image

**Pull request merged**